packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3512,6 +3595,24 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
   BuildOwner? get owner => _owner;
   BuildOwner? _owner;
 
+  BuildScope? _buildScope;
+  /// A [BuildScope] within which the [Element] tree should be updated together.
+  ///
+  /// The default implementation returns the parent [Element]'s [buildScope],
+  /// as in most cases an [Element] is ready to rebuild as soon as its ancestors
+  /// are no longer dirty. One notable exception is [LayoutBuilder]'s
+  /// descendants, which must not rebuild until the incoming constraints become
+  /// available. [LayoutBuilder]'s [Element] overrides [buildScope] to make none
+  /// of its descendants can rebuild until the incoming constraints are known.
+  ///
```

Comment:
Somewhere this should include a very clear warning that when you override buildScope, your subtree will not rebuild automatically and you are in charge of manually calling `owner.buildScope` for your element!